corehq/apps/app_manager/models.py

```diff
@@ -2767,10 +2767,21 @@ def validate_for_build(self):
             forms = self.forms
 
             case_tag = None
+            loaded_case_types = None
             for form in forms:
                 info = self.get_module_info()
                 form_info = {"id": form.id if hasattr(form, 'id') else None, "name": form.name}
                 non_auto_select_actions = [a for a in form.actions.load_update_cases if not a.auto_select]
+                this_forms_loaded_case_types = {action.case_type for action in non_auto_select_actions}
+                if loaded_case_types is None:
+                    loaded_case_types = this_forms_loaded_case_types
+                elif loaded_case_types != this_forms_loaded_case_types:
```

Comment:
This seems stricter than we need it to be (though that may be ok). My impression is that as long as (1) they all load case list screen defined by that module (the one that has the case list form) and (2) the path leading up to that screen is the same, then it'll build. (I'm not even totally certain we need (2)). Is that right? I'm not sure I understand this super well, but that was my impression looking at the error a few days ago.